### PR TITLE
refactor(object store): unify s3 compatible storage and s3 backend

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
As s3-compatible storage is almost the same with s3, the only different is to provide the endpoint, so this PR remove s3 compatible mode.

In the future, if user want to use s3-compatible storage, he need to config 
```
export AWS_REGION=your_region
export AWS_ACCESS_KEY_ID=your_access_key
export AWS_SECRET_ACCESS_KEY=your_secret_key
export OBJECT_STORAGE_ENDPOINT=your_endpoint 
```

**Before**
Passing `hummock+s3-compatible://bucket-name`

**Now**
Passing `hummock+s3://bucket-name`

I think there will be some changes in RisingWave Operator and RisingWave Test.

## Checklist For Contributors

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [ ] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by arkbriar** [comment]
Generally LGTM. However please make it compatible for at least one release when introducing a breaking change. That is to say, leave the `hummock://s3-compatible` interface for now to let the downstream projects have a chance to change.

**Comment by arkbriar** [comment]
Looks like there are some conflicts, others LGTM.

**Comment by wcy-fdu** [comment]
@hzxa21  PTAL❤️

File: docker/aws.env [hunk 1]
```diff
@@ -1,13 +1,13 @@
 RUST_BACKTRACE="1"
 
 # The region of your object storage bucket
-S3_COMPATIBLE_REGION=your_region
+AWS_REGION=your_region
 
 # The endpoint URL for your object storage
-S3_COMPATIBLE_ENDPOINT=your_endpoint
+OBJECT_STORAGE_ENDPOINT=your_endpoint
```
Comment (hzxa21) [inline docker/aws.env:7]: minor: This env is intended to be used only for S3 enpoint. I think `RW_S3_ENDPOINT` is a better naming.

File: src/object_store/src/object/s3.rs [hunk 2]
```diff
@@ -533,53 +533,19 @@ impl S3ObjectStore {
...
-                access_key_secret,
-                None,
-            ))
-            .endpoint_url(endpoint)
-            .load()
-            .await;
+        let sdk_config_loader =
+            aws_config::from_env().retry_config(RetryConfig::standard().with_max_attempts(4));
+        let endpoint = std::env::var("OBJECT_STORAGE_ENDPOINT").unwrap_or("".to_string());
+        let sdk_config = match endpoint.is_empty() {
```
Comment (hzxa21) [inline src/object_store/src/object/s3.rs:539]: nits: we can match `std::env::var("OBJECT_STORAGE_ENDPOINT")` directly and combine this two lines

File: src/object_store/src/object/mod.rs [hunk 3]
```diff
@@ -738,19 +730,20 @@ pub async fn parse_remote_object_store(
...
-                        .strip_prefix("s3-compatible://")
-                        .unwrap()
-                        .to_string(),
-                    metrics.clone(),
-                )
-                .await
-                .monitored(metrics),
+
+        s3_compatible if s3_compatible.starts_with("s3-compatible://") => ObjectStoreImpl::S3(
+            // For backward compatibility, s3-compatible is still reserved.
```
Comment (hzxa21) [inline src/object_store/src/object/mod.rs:735]: Do we want100% compatibility? If so, we also need to set the new envs if the old envs are set.

File: src/object_store/src/object/s3.rs [hunk 4]
```diff
@@ -532,54 +532,45 @@ impl S3ObjectStore {
     ///
     /// See [AWS Docs](https://docs.aws.amazon.com/sdk-for-rust/latest/dg/credentials.html) on how to provide credentials and region from env variable. If you are running compute-node on EC2, no configuration is required.
     pub async fn new(bucket: String, metrics: Arc<ObjectStoreMetrics>) -> Self {
-        // Retry 3 times if we get server-side errors or throttling errors
-        let sdk_config = aws_config::from_env()
-            .retry_config(RetryConfig::standard().with_max_attempts(4))
-            .load()
-            .await;
-        let client = Client::new(&sdk_config);
+        // The following code is for compatibility.
```
Comment (wcy-fdu) [inline src/object_store/src/object/s3.rs:535]: Will remove these after all downstream project finish changes.

**Review by hzxa21 — APPROVED** [review]

**Pull request merged**